android/libraries/rib-base/src/main/java/com/badoo/ribs/core/routing/configuration/ConfigurationCommandCreator.kt

```diff
@@ -0,0 +1,205 @@
+package com.badoo.ribs.core.routing.configuration
+
+import android.os.Parcelable
+import com.badoo.ribs.core.routing.configuration.ConfigurationCommand.SingleConfigurationCommand.Activate
+import com.badoo.ribs.core.routing.configuration.ConfigurationCommand.SingleConfigurationCommand.Add
+import com.badoo.ribs.core.routing.configuration.ConfigurationCommand.SingleConfigurationCommand.Deactivate
+import com.badoo.ribs.core.routing.configuration.ConfigurationCommand.SingleConfigurationCommand.Remove
+import com.badoo.ribs.core.routing.configuration.ConfigurationKey.Content
+import com.badoo.ribs.core.routing.configuration.ConfigurationKey.Overlay
+import com.badoo.ribs.core.routing.configuration.ConfigurationKey.Overlay.Key
+import com.badoo.ribs.core.routing.configuration.feature.BackStackElement
+import com.badoo.ribs.core.routing.configuration.feature.BackStackFeature
+import io.reactivex.Observable
+import java.lang.Math.min
+
+/**
+ * Takes the state emissions from [BackStackFeature], and translates them to a stream of
+ * [ConfigurationCommand]s.
+ *
+ * @see [ConfigurationCommandCreator.diff]
+ */
+internal fun <C : Parcelable> BackStackFeature<C>.commands(): Observable<ConfigurationCommand<C>> =
+    Observable.wrap(this)
+        .startWith(initialState) // Bootstrapper can overwrite it by the time we receive the first state emission here
```

Comment:
I don't fully understand why we need this. Looks like feature emits initial state in case of restoring from bundle, how bootstrapper can overwrite it?

Reply:
It follows from how MVICore works. You construct a `Feature`, it immediately invokes `Bootstrapper`, which can result in an immediate new state emission. By the time you have the chance to subscribe to your `Feature`, it's last state is already the one emitted by the `Bootstrapper`, and its initial empty state is no longer accessable. In this case above I need to convert the transition from empty -> state created by `Bootstrapper` to actions too, otherwise the initial configuration wouldn't get attached to the screen.